### PR TITLE
Zip Extensions

### PR DESCRIPTION
**Pull request by awestover**
Goal: 
Upload a minimal set of files to S3 to enable testing extension downloading on staging. 
In particular, we need to create and upload the following files:

```
v14/ext_index.json
    -- this contains information necessary to create control files
v14/extensions/embedding.tar.gz
    -- this contains the library files and sql files necessary to create this extension
v14/extensions/anon.tar.gz
	-- it's structure when unzipped is `/extension/share`, `/extension/lib`
```

`ext_index.json` should look like this:
```json
{
  "enabled_extensions": {
    "123454321": [
      "anon"
    ],
    "public": [
      "embedding"
    ]
  },
  "control_data": {
    "embedding": "comment = 'hnsw index' \ndefault_version = '0.1.0' \nmodule_pathname = '$libdir/embedding' \nrelocatable = true \ntrusted = true",
    "anon": "# PostgreSQL Anonymizer (anon) extension \ncomment = 'Data anonymization tools' \ndefault_version = '1.1.0' \ndirectory='extension/anon' \nrelocatable = false \nrequires = 'pgcrypto' \nsuperuser = false \nmodule_pathname = '$libdir/anon' \ntrusted = true \n"
  }
}
```

### CODE REVIEW

File: Dockerfile.compute-node [hunk 1]
```diff
@@ -753,7 +753,7 @@ FROM scratch AS postgres-extensions
 # COPY --from=postgres-cleanup-layer /usr/local/pgsql/lib             /usr/local/pgsql/lib
 # Custom extensions
 COPY --from=pg-anon-pg-build /extensions/anon/lib/ /extensions/anon/lib
-COPY --from=pg-anon-pg-build /extensions/anon/share/extension /extensions/anon/share/extension
+RUN --from=pg-anon-pg-build tar -zcvf /extensions/anon/share/extension/anon.tar.gz /extensions/anon/share/extension
```
Comment (bayandin) [inline Dockerfile.compute-node:756]: This probably doesn't do what we want.
We need to create an archive in `pg-anon-pg-build` step and COPY it to the extension directory here.

And don't we want to add `libs` to the archive as well, or am I missing something?

Comment (awestover) [reply Dockerfile.compute-node]: ah yes, I've hopefully fixed this issue.

Comment (awestover) [reply Dockerfile.compute-node]: let me know if it's still broken

**Comment by awestover** [comment]
This PR is not actually blocking testing custom extensions on staging. 
So I'm de-prioritizing it for the moment

**Comment by awestover** [comment]
closed in favor of https://github.com/neondatabase/neon/pull/4764

**Pull request closed without merging**